xbmc/cores/DllLoader/exports/emu_msvcrt.cpp

```diff
@@ -999,8 +999,8 @@ extern "C"
         strncpy(entry->d_name, "..\0", 3);
       else
       {
-        strncpy(entry->d_name, dirData->items[dirData->curr_index - 2]->GetLabel().c_str(), sizeof(entry->d_name));
-        entry->d_name[sizeof(entry->d_name)-1] = '\0'; // null-terminate any truncated paths
+        snprintf(entry->d_name, sizeof(entry->d_name), "%s",
```

Comment:
You can use `strncpy_s` to make sure not introduce unwanted behavior change

```suggestion
        strncpy_s(entry->d_name, sizeof(entry->d_name),
                  dirData->items[dirData->curr_index - 2]->GetLabel().c_str(),
                  sizeof(entry->d_name));
```

Reply:
I don't think that's supported on all platforms is it?  GCC doesn't appear to have it

Reply:
I had doubts about the first fragment because it seems that the original code contemplates that the strings are not null terminated but it seems unlikely.